docs/codeql/codeql-language-guides/annotations-in-java.rst

```diff
@@ -213,11 +213,11 @@ To do so, we first introduce a class for representing all ``@SuppressWarnings``
    class SuppressDeprecationWarningAnnotation extends Annotation {
        SuppressDeprecationWarningAnnotation() {
            this.getType().hasQualifiedName("java.lang", "SuppressWarnings") and
-           this.getAValue().(Literal).getLiteral().regexpMatch(".*deprecation.*")
+           this.getAStringArrayValue("value").regexpMatch(".*deprecation.*")
        }
    }
 
-Here, we use ``getAValue()`` to retrieve any annotation value: in fact, annotation type ``SuppressWarnings`` only has a single annotation element, so every ``@SuppressWarnings`` annotation only has a single annotation value. Then, we ensure that it is a literal, obtain its string value using ``getLiteral``, and check whether it contains the string ``deprecation`` using a regular expression match.
+Here, we use ``getAStringArrayValue("value")`` to retrieve any of the suppressed warnings: ``@SuppressWarnings`` defines the warnings to suppress using the annotation element named ``value`` of type ``String[]``, and ``getAStringArrayValue`` retrieves all of the array values; the CodeQL class ``Annotation`` also has similar convenience predicates for the other possible annotation element types. Afterwards we check whether one of the values is the string ``deprecation`` using a regular expression match.
 
 For real-world use, this check would have to be generalized a bit: for example, the OpenJDK Java compiler allows ``@SuppressWarnings("all")`` annotations to suppress all warnings. We may also want to make sure that ``deprecation`` is matched as an entire word, and not as part of another word, by changing the regular expression to ``".*\\bdeprecation\\b.*"``.
```

Comment:
Side note: Not completely sure why a regex match is performed here in the first place. Maybe the original author thought the suppressions were a single `String` instead of a `String[]`?